manager/status_handler.py

```diff
@@ -50,114 +56,183 @@ class PatchStatus(PatchRequest):
         else:
             # change status for all systems for given CVE(s)
             in_inventory_id_list = None
 
         in_cve_list = parse_str_or_list(data["cve"])
+        in_status_id = data.get("status_id")
+        if data.get("status_text") and data["status_text"].strip():
+            in_status_text = data["status_text"].strip()
+        else:
+            in_status_text = None
+
+        return in_inventory_id_list, in_cve_list, in_status_id, in_status_text
 
-        status_to_cves_map = {}
-        status_text_to_cves_map = {}
-        if "status_id" in data:
-            # single status for all CVEs
-            status_to_cves_map[data["status_id"]] = in_cve_list
-        if "status_text" in data:
-            # single status for all CVEs
-            try:
-                key = data["status_text"].strip() if data["status_text"].strip() else None
-            except AttributeError:
-                key = None
-            status_text_to_cves_map[key] = in_cve_list
-        # if neither of status_id or status_text is set => inherit from CVE-level
-        if not status_to_cves_map and not status_text_to_cves_map:
-            # use CVE-level status if status not specified
-            cve_details = (CveAccountData.select(CveMetadata.cve, CveAccountData.status_id, CveAccountData.status_text)
-                           .join(CveMetadata, on=(CveAccountData.cve_id == CveMetadata.id))
-                           .where((CveAccountData.rh_account_id == rh_account_id) &
-                                  (CveMetadata.cve << in_cve_list))
+    @staticmethod
+    def _apply_system_list_filter(query, rh_account_id, in_inventory_id_list):
+        query = cyndi_join(query)
+        query = query.where((SystemPlatform.rh_account_id == rh_account_id) &
+                            (SystemPlatform.when_deleted.is_null(True)))
+        if in_inventory_id_list is not None:
+            query = query.where(SystemPlatform.inventory_id << in_inventory_id_list)
+        return query
+
+    @classmethod
+    def _get_current_status(cls, rh_account_id, in_inventory_id_list, in_cve_list):
+        # pair status
+        system_cve_details = (SystemCveData.select(SystemPlatform.inventory_id, CveMetadata.cve,
+                                                   SystemCveData.status_id, SystemCveData.status_text)
+                              .join(CveMetadata, on=(SystemCveData.cve_id == CveMetadata.id))
+                              .join(SystemPlatform, on=(SystemCveData.system_id == SystemPlatform.id))
+                              .where(CveMetadata.cve << in_cve_list)
+                              .dicts())
+        system_cve_details = cls._apply_system_list_filter(system_cve_details, rh_account_id, in_inventory_id_list)
+        current_status = {}
+        for system_cve_detail in system_cve_details:
+            current_status.setdefault(system_cve_detail["cve"], {})[system_cve_detail["inventory_id"]] = \
```

Comment:
You will hate me for this 🙈, this could be transformed to `NamedTuple` so it can be more readable. Also maybe types for functions wouldn't be bad.